contract/src/launchIt.js

```diff
@@ -0,0 +1,227 @@
+ * @param {import('@agoric/vat-data').Baggage} _baggage
+ */
+export const start = async (zcf, _privateArgs, _baggage) => {
+  const { name, supplyQty, displayInfo = {}, brands } = zcf.getTerms();
+  mustMatch(brands, M.splitRecord({ Deposit: BrandShape }));
+  const { zcfSeat: deposits } = zcf.makeEmptySeatKit();
+
+  const assetMint = await zcf.makeZCFMint(kw(name), AssetKind.NAT, displayInfo);
+  const asset = assetMint.getIssuerRecord();
+  const share = await makeZcfIssuerKit(zcf, 'Share');
+
+  const ShapeAmt = {
+    Asset: harden({ brand: asset.brand, value: M.nat() }),
+    Deposit: harden({ brand: brands.Deposit, value: M.nat() }),
+    Share: harden({ brand: share.brand, value: M.nat() }),
+  };
+  const ExitDeadlineShape = {
+    afterDeadline: {
+      timer: TimerServiceShape,
+      deadline: TimestampRecordShape,
+    },
+  };
+  // export these from a client interface module?
+  const Shape = harden({
+    Amount: ShapeAmt,
+    Proposal: {
+      Mint: M.splitRecord({
+        exit: ExitDeadlineShape,
+      }),
+      Collect: M.splitRecord(
+        { exit: ExitDeadlineShape },
+        { want: { Deposit: ShapeAmt.Deposit } },
+      ),
+      Deposit: M.splitRecord(
+        { give: { Deposit: ShapeAmt.Deposit } },
+        { want: { Shares: ShapeAmt.Share } },
+      ),
+      Withdraw: M.splitRecord({
+        want: { Deposit: ShapeAmt.Deposit },
+        give: { Shares: ShapeAmt.Share },
+      }),
+      Redeem: M.splitRecord(
+        { give: { Shares: ShapeAmt.Share } },
+        { want: { Minted: ShapeAmt.Asset } },
+      ),
+    },
+  });
+
+  /** @type {(Proposal & { exit: AfterDeadlineExitRule })| undefined} */
+  let mintProposal;
+  const getMintProposal = () => mintProposal || Fail`not yet minted`;
+  /** @type {Amount<'nat'> | undefined} */
+  let totalShares;
+  const assertCanWithdraw = () =>
+    !totalShares || Fail`lauch committed; shares can only be redeemed`;
+  const getLaunchResult = () =>
+    mintProposal && totalShares
+      ? { Minted: AmountMath.make(asset.brand, supplyQty), Shares: totalShares }
+      : Fail`not launching`;
+
+  const { zcfSeat: stage } = zcf.makeEmptySeatKit();
+
+  /** @type {OfferHandler} */
+  const depositHandler = subscriber => {
+    getMintProposal();
+    const { give } = subscriber.getProposal();
+    const { Deposit } = give;
+    const Shares = AmountMath.make(share.brand, Deposit.value);
+    stage.clear();
+    share.mint.mintGains({ Shares }, stage);
+    atomicRearrange(
+      zcf,
+      harden([
+        [subscriber, deposits, give],
+        [stage, subscriber, { Shares }],
+      ]),
+    );
+    subscriber.exit();
+    return true;
+  };
+
+  /** @type {OfferHandler} */
+  const withdrawHandler = subscriber => {
+    assertCanWithdraw();
+    const { want, give } = subscriber.getProposal();
+    const { Shares } = give;
+    const { Deposit } = want;
+    mustMatch(Shares.value, M.gte(Deposit.value), 'insufficient shares');
```

Comment:
Is this essentially an assert? Should it be named so in order to be more familiar?

Reply:
yes, it's an assert. To my mind, `mustMatch` says that sufficiently clearly.